jobs/build-cosa.Jenkinsfile

```diff
@@ -44,6 +44,22 @@ properties([
     durabilityHint('PERFORMANCE_OPTIMIZED')
 ])
 
+node {
+    change = checkout(
+        changelog: false,
+        poll: false,
+        scm: [
+            $class: 'GitSCM',
+            branches: [[name: 'origin/rhcos*'], [name: 'origin/main']],
+            userRemoteConfigs: [[url: 'https://github.com/coreos/coreos-assembler.git']],
+            extensions: [[$class: 'CloneOption',
+                          noTags: true,
+                          reference: '', shallow: true]]
+        ]
+    )
+    params.COREOS_ASSEMBLER_GIT_REF = change.GIT_BRANCH['origin/'.length()..-1]
```

Comment:
We might as well `git rev-parse` the ref here into `commit` to avoid races between webhook events and `ls-remote` below.